### ISSUE TITLE
Creating CLI for managing Permify instances 

### ISSUE THREAD

**Issue by EgeAytin** [Permify/permify#146]

**Comment by hungaikev** [comment]
Can I Attempt this issue? Are there specific CLI libraries that are preferred?

**Comment by hungaikev** [comment]
attempt

**Comment by b4s36t4** [comment]
@tolgaOzen Just wondering that CLI should be in golang as well?

**Comment by sakshambp** [comment]
can the cli be in python? if yes i can contribute Please assign

**Comment by uasouz** [comment]
@tolgaOzen Does this bounty still up and is someone working on it? If it still up the issue will be moved to the CLI repository? And if no one is already working on it can I work on it?

**Comment by rogueloop** [comment]
/attempt [#146](https://github.com/Permify/permify/issues/146)

**Comment by swaingotnochill** [comment]
@EgeAytin Is this still up for grab?

**Comment by harshsbhat** [comment]
Sounds interesting tbh. Always wanted to build a CLI for prod.

**Comment by ashish111333** [comment]
is this still open ? @EgeAytin

**Comment by harshsbhat** [comment]
@ashish111333 I don't think so. If you open the docs for permify. You can see they already have a CLI

**Comment by ashish111333** [comment]
> @ashish111333 I don't think so. If you open the docs for permify. You can see they already have a CLI
Thanks, I think the maintainers should make it clearthat "bounty is finished", so that any newcomer doesn't start working on it.

**Comment by harshsbhat** [comment]
@EgeAytin Can you close this issue?

**Comment by aadarsh-nagrath** [comment]
yeah close it

**Comment by Wolf-06** [comment]
is it still open ?

**Comment by jacanidjar** [comment]
is this still open ? @EgeAytin

**Comment by naodEthiop** [comment]
Hello! 👋

I’m interested in working on the CLI development for managing Permify instances. The objective of creating a command-line tool to simplify interaction with Permify sounds great, and I would like to contribute to implementing it.

Before starting, I wanted to ask a couple of questions to ensure the implementation matches the project’s expectations:

• Is there a preferred CLI framework (for example Cobra) that should be used?

• Are there specific commands or features you would like the first version of the CLI to include?

• Should the CLI interact with Permify primarily through its API, Docker commands, or another interface?

If the issue is still open, I would be happy to take it and start working on a proposal or initial implementation.

Thank you!

**Comment by CuboYe** [comment]
/claim 146

Claiming territory for a scoped fix attempt now. If maintainer preference differs, I can adapt quickly.

**Comment by CuboYe** [comment]
/claim 146

Claiming this bounty issue. I reviewed the scope and will break it into a concrete implementation plan (commands, config persistence, and instance lifecycle operations) before opening the PR.

**Comment by thisisnkc** [comment]
Hey everyone! Wanted to share something that might be relevant to this discussion.

I've been building **[permify-toolkit](https://github.com/thisisnkc/permify-toolkit)** - a type-safe TypeScript toolkit for Permify that already includes a CLI package (`@permify-toolkit/cli`). It might scratch some of the itches being discussed here, especially for TypeScript/Node.js teams.

**What the CLI can do right now:**

- `schema push` - Push your Permify schema to the server from a typed config file
- `schema diff` - Preview exactly what will change before a push (added/removed/modified entities, relations, permissions) - great for CI pipelines with `--exit-code` flag
- `schema validate` - Validate your schema locally before touching the server
- `relationships seed` - Seed relationships from a JSON file (supports append or replace modes)
- `relationships list` / `relationships export` - List and export existing relationships

The whole thing is driven by a single `permify.config.ts` file - so your schema, endpoint, tenant, and seed data are all defined in one place. No scattered flags or env var juggling.

**Why I built it this way:**

Working with Permify's gRPC API directly meant a lot of boilerplate every time I needed to push a schema change or check what was deployed. The `schema diff` command in particular has saved me a few times from accidentally nuking a production schema - it shows a structured entity-level diff before you commit to a push.

The repo also includes a **[nestjs](https://github.com/thisisnkc/permify-toolkit/tree/main/packages/nestjs)** module and a **[core](https://github.com/thisisnkc/permify-toolkit/tree/main/packages/core)** package if you're integrating Permify into an existing Node backend.

Would love feedback from folks in this thread since you're clearly thinking about the same problems. Happy to discuss what's missing or where the official CLI could take things further that a community tool can't.

---

GitHub: https://github.com/thisisnkc/permify-toolkit
